### ISSUE TITLE
Allow a Moshi instance to not use generated adapters

### ISSUE THREAD

**Issue by edenman** [square/moshi#761]
I'm in the process of migrating from the old Java reflection adapters to the new Kotlin adapters.  My data model has a bunch of dark nasty corners where things are specified as non-nullable, but the server might actually not return them in some cases.  I'd like to start using the new codegen adapters, but fall back to the Java reflection adapters (which allow these sorts of inconsinstencies) in case there's an exception.  Once I whack-a-mole all of these instances in production and fix all the issues in my data model, I can just migrate to using only the codegen adapters.

This works fine with `KotlinJsonAdapterFactory` since I can build one Moshi instance with it and try that one, then fall back to another Moshi instance (sans `KotlinJsonAdapterFactory`) if the Kotlin one fails.  However, the codegen adapters seem to be built into StandardJsonAdapters and there's no way to opt out of those adapters if the model classes are annotated+generated.

It'd be rad if there was an opt-out flag.  Or if the generated adapters were moved to their own `GeneratedKotlinJsonAdapterFactory` and it was made an opt-in, although I can understand not wanting to do that.

**Comment by edenman** [comment]
Not the end of the world if this isn't possible since I'm only planning on having a couple releases with this bridge code in them, but I'd like to avoid taking the perf hit of kotlin-reflect (both in APK size and in parsing perf):
```
 Java reflection for class GetMapStateResponse took 872ms
 Kotlin reflection for class GetMapStateResponse took 17662ms
```

**Comment by edenman** [comment]
It's way worse for some other responses 😭 
```
Java reflection for class GetEarlyAccessRegionsAndWaitlistInfoResponse took 77ms 
Kotlin reflection for class GetEarlyAccessRegionsAndWaitlistInfoResponse took 1748ms
```

**Comment by swankjesse** [comment]
Whoa, those timing numbers are absurdly bad!

**Comment by edenman** [comment]
Yeah I use a _really_ shitty phone for testing.  Keeps me honest about perf.

**Comment by NightlyNexus** [comment]
There's not a great workaround for this. A possible hack would be to install your own ClassJsonAdapter factory to take precendence over the codegen.

The next release will make this moot, since it will fail early for Kotlin code that doesn't use custom adapters, the KotlinJsonAdapterFactory, or the Kotlin codegen adapters. So, I can't think of what Moshi can help with here.
